contracts/interfaces/IRamm.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity >=0.5.0;
+
+import "./IPool.sol";
+import "./INXMToken.sol";
+import "./ITokenController.sol";
+
+
+enum UintParams {
+  targetLiquidity,
+  twapDuration,
+  aggressiveLiqSpeed,
+  oracleBuffer
+}
+
+struct Pool {
+  uint96 nxmReserve;
+  uint16 liquiditySpeed;
+  uint16 ratchetSpeed;
+}
+
+struct Observation {
+  uint32 timestamp;
+  uint64 priceCumulativeAbove;
+  uint64 priceCumulativeBelow;
+}
+
+ struct CumulativePriceCalculationProps {
+  uint previousEthReserve;
+  uint currentEthReserve;
+  uint previousNxmA;
+  uint currentNxmA;
+  uint previousNxmB;
+  uint currentNxmB;
+  uint ratchetSpeedA;
+  uint ratchetSpeedB;
```

Comment:
is this always the current ratchet speed?

Reply:
For now yes, but as Rei mentioned yesterday we will add festRatchetSpeed for below pool